packages/features/bookings/Booker/Booker.tsx

```diff
@@ -0,0 +1,196 @@
+        )}>
+        <AnimatePresence>
+          <StickyOnDesktop key="meta">
+            <BookerSection area="meta">
+              <EventMeta event={event.data} isLoading={event.isLoading} selectedTime={selectedTimeslot} />
+              {layout !== "small_calendar" && (
+                <div className=" mt-auto p-6">
+                  <DatePicker onDaySelect={onDaySelect} onMonthChange={onMonthChange} />
+                </div>
+              )}
+            </BookerSection>
+          </StickyOnDesktop>
+
+          <BookerSection
+            key="book-event-form"
+            area="main"
+            className="dark:border-darkgray-300 sticky top-0 ml-[-1px] h-full border-gray-200 p-6 md:border-l"
+            {...fadeInUp}
+            visible={bookerState === "booking"}>
+            <BookEventForm onCancel={() => setSelectedTimeslot(null)} />
+          </BookerSection>
+
+          <BookerSection
+            key="datepicker"
+            area="main"
+            visible={bookerState !== "booking" && layout === "small_calendar"}
+            {...fadeInUp}
+            initial="visible"
+            className="md:dark:border-darkgray-300 ml-[-1px] h-full p-6 md:border-l md:border-gray-200">
+            <DatePicker onDaySelect={onDaySelect} onMonthChange={onMonthChange} />
+          </BookerSection>
+
+          <BookerSection
+            key="large-calendar"
+            area="main"
+            visible={
+              layout === "large_calendar" &&
+              (bookerState === "selecting_date" || bookerState === "selecting_time")
+            }
+            className="dark:border-darkgray-300 sticky top-0  ml-[-1px] h-full border-gray-200 md:border-l"
+            {...fadeInUp}>
+            <LargeCalendar onDaySelect={onDaySelect} onTimeSelect={onTimeSelect} />
+          </BookerSection>
+
+          <BookerSection
+            key="timeslots"
+            area={{ default: "main", small_calendar: "timeslots" }}
+            visible={
+              (bookerState === "selecting_time" && layout === "small_calendar") ||
+              (layout === "large_timeslots" && bookerState !== "booking")
+            }
+            className={classNames(
+              "dark:border-darkgray-300 flex w-full flex-row border-gray-200 p-6 pb-0 md:border-l",
+              layout === "small_calendar" && "h-full overflow-auto",
+              layout !== "small_calendar" && "sticky top-0"
+            )}
+            {...fadeInLeft}>
+            <AvailableTimeSlots
+              extraDays={layout === "large_timeslots" ? 4 : 0}
+              onTimeSelect={onTimeSelect}
+              limitHeight={layout === "small_calendar"}
+              seatsPerTimeslot={event.data?.seatsPerTimeSlot}
+            />
+          </BookerSection>
+        </AnimatePresence>
+      </m.div>
+    </>
+  );
+};
+
+export const Booker = (props: BookerProps) => {
+  if (props.isAway) return <Away />;
```

Comment:
This away state is passed into the booker component. This can be either done client side, but in our own case we already fetch this boolean server side, so we can show the away state asap without running api calls client side.